src/main/webapp/app/tutor-exercise-dashboard/tutor-exercise-dashboard.component.ts

```diff
@@ -261,6 +261,15 @@ export class TutorExerciseDashboardComponent implements OnInit {
         return 'DRAFT';
     }
 
+    textSubmissionLanguageTranslationKey(submission: Submission): string {
+        if (submission.submissionExerciseType !== SubmissionExerciseType.TEXT) {
+            throw new Error('Cannot determine text submission language for non-text-exercise.');
```

Comment:
The exception is only needed because it's a method. If you do the work in the template, you don't need this method.